analyzers/src/SonarAnalyzer.Common/Rules/ToStringShouldNotReturnNullBase.cs

```diff
@@ -63,5 +63,9 @@ protected void ToStringReturnsNull(SyntaxNodeAnalysisContext context, SyntaxNode
         node.Ancestors()
             .TakeWhile(x => !IsLocalOrLambda(x))
             .Any(x => Language.Syntax.IsKind(x, MethodKind)
-                && nameof(ToString).Equals(Language.Syntax.NodeIdentifier(x)?.ValueText, Language.NameComparison));
+                && nameof(ToString).Equals(Language.Syntax.NodeIdentifier(x)?.ValueText, Language.NameComparison)
+                && !IsStatic(x));
+
+    private bool IsStatic(SyntaxNode node) =>
+        Language.Syntax.IsStatic(node);
```

Comment:
Why have this method when you can call directly the method from the SyntaxFacade?